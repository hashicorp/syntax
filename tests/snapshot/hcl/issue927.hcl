variable "foo" {}
output "result-val" { value = var.foo }

variable "some-var" {
  default = "value"
}

module "foo-mod" {
  source = "./foo"
  foo = var.some-var
}

module "bar" {
  source = "./foo"
  foo = module.foo-mod.result-val
}