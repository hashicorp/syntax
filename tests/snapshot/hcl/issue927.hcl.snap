>variable "foo" {}
#^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#        ^ source.hcl meta.block.hcl
#         ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#              ^ source.hcl meta.block.hcl
#               ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>output "result-val" { value = var.foo }
#^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#      ^ source.hcl meta.block.hcl
#       ^^^^^^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                   ^ source.hcl meta.block.hcl
#                    ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                     ^ source.hcl meta.block.hcl
#                      ^^^^^ source.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#                           ^ source.hcl meta.block.hcl variable.declaration.hcl
#                            ^ source.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#                             ^ source.hcl meta.block.hcl variable.declaration.hcl
#                              ^^^ source.hcl meta.block.hcl
#                                 ^ source.hcl meta.block.hcl keyword.operator.accessor.hcl
#                                  ^^^ source.hcl meta.block.hcl variable.other.member.hcl
#                                     ^ source.hcl meta.block.hcl
#                                      ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>variable "some-var" {
#^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#        ^ source.hcl meta.block.hcl
#         ^^^^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                   ^ source.hcl meta.block.hcl
#                    ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>  default = "value"
#^^ source.hcl meta.block.hcl
#  ^^^^^^^ source.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#         ^ source.hcl meta.block.hcl variable.declaration.hcl
#          ^ source.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#           ^ source.hcl meta.block.hcl variable.declaration.hcl
#            ^ source.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#             ^^^^^ source.hcl meta.block.hcl string.quoted.double.hcl
#                  ^ source.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>module "foo-mod" {
#^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#      ^ source.hcl meta.block.hcl
#       ^^^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                ^ source.hcl meta.block.hcl
#                 ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>  source = "./foo"
#^^ source.hcl meta.block.hcl
#  ^^^^^^ source.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#        ^ source.hcl meta.block.hcl variable.declaration.hcl
#         ^ source.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#          ^ source.hcl meta.block.hcl variable.declaration.hcl
#           ^ source.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#            ^^^^^ source.hcl meta.block.hcl string.quoted.double.hcl
#                 ^ source.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>  foo = var.some-var
#^^ source.hcl meta.block.hcl
#  ^^^ source.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#     ^ source.hcl meta.block.hcl variable.declaration.hcl
#      ^ source.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#       ^ source.hcl meta.block.hcl variable.declaration.hcl
#        ^^^ source.hcl meta.block.hcl
#           ^ source.hcl meta.block.hcl keyword.operator.accessor.hcl
#            ^^^^^^^^ source.hcl meta.block.hcl variable.other.member.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>module "bar" {
#^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#      ^ source.hcl meta.block.hcl
#       ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#            ^ source.hcl meta.block.hcl
#             ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>  source = "./foo"
#^^ source.hcl meta.block.hcl
#  ^^^^^^ source.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#        ^ source.hcl meta.block.hcl variable.declaration.hcl
#         ^ source.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#          ^ source.hcl meta.block.hcl variable.declaration.hcl
#           ^ source.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#            ^^^^^ source.hcl meta.block.hcl string.quoted.double.hcl
#                 ^ source.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>  foo = module.foo-mod.result-val
#^^ source.hcl meta.block.hcl
#  ^^^ source.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#     ^ source.hcl meta.block.hcl variable.declaration.hcl
#      ^ source.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#       ^ source.hcl meta.block.hcl variable.declaration.hcl
#        ^^^^^^ source.hcl meta.block.hcl
#              ^ source.hcl meta.block.hcl keyword.operator.accessor.hcl
#               ^^^^^^^ source.hcl meta.block.hcl variable.other.member.hcl
#                      ^ source.hcl meta.block.hcl keyword.operator.accessor.hcl
#                       ^^^^^^^^^^ source.hcl meta.block.hcl variable.other.member.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl